### PR TITLE
Velocity smoother ros2 port

### PR DESCRIPTION
**Pull request by clalancette**
A port of the velocity smoother to ROS 2.  A few limitations:

1.  Compile tested against Eloquent
1.  Tested to launch in a simple way
1.  Not tested as part of a full application
1.  Launch files not ported
1.  Tests not ported
1.  Dynamic changing of parameters not yet supported

This PR also adds empty COLCON_IGNORE files for all of the pieces that don't yet compile against ROS 2.

### CODE REVIEW

File: kobuki_velocity_smoother/include/kobuki_velocity_smoother/velocity_smoother.hpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/**
+ * @file /include/yocs_velocity_smoother/velocity_smoother_nodelet.hpp
+ *
+ * @brief Velocity smoother interface
+ *
+ * License: BSD
+ *   https://raw.github.com/yujinrobot/yujin_ocs/hydro/yocs_velocity_smoother/LICENSE
```
Comment (stonier) [inline kobuki_velocity_smoother/include/kobuki_velocity_smoother/velocity_smoother.hpp]: Licenses ... this repo generally handles it by copy pasting the content up front here. Should follow that.

**Review by stonier — APPROVED** [review]
Pending license update, this looks good to go - albeit partial.

File: kobuki_velocity_smoother/include/kobuki_velocity_smoother/velocity_smoother.hpp (hunk 1 above)
Comment (clalancette) [reply kobuki_velocity_smoother/include/kobuki_velocity_smoother/velocity_smoother.hpp]: Fixed in ad27a3a

**Comment by clalancette** [comment]
I ended up doing some work here to make the parameters configurable on-the-fly, fixing up any threading issues, and porting the launchfiles to ROS 2.  @stonier I think this is good to go now, so feel free to merge at your convenience.

**Comment by stonier** [comment]
You know, might have been nice for the history to get the original code in, then the changes so we can do a diff across commits. Let's talk about that tomorrow, we can probably still easily achieve that.

**Comment by clalancette** [comment]
> You know, might have been nice for the history to get the original code in, then the changes so we can do a diff across commits. Let's talk about that tomorrow, we can probably still easily achieve that.

The commit history is actually setup like that here.  Commit 76f068fd826713de23436bd9d3f34e31cfcdb5df is a direct copy of the original code, and then the follow-on commits actually do the port.

That all being said, after looking at this package, I'm almost thinking we should make it a separate package.  The functionality is really pretty generic, and not tied to kobuki at all, so we could easily do that.

**Comment by stonier** [comment]
Aye, I had those two thoughts (interactively rebase these commits & wish for a separate package) as I hit the hay last night but couldn't be bothered to immediately get back to the desk and post them!

Next steps:

* I'll shift over this package as is (sans this PR) to https://github.com/kobuki-base/kobuki_ros and COLCON_IGNORE everything bar the metapackage. Then I'll release and make sure we have webhooks working so we can incrementally unignore everything with confidence.
* New home for you to move the velocity smoother to at https://github.com/kobuki-base/velocity_smoother.

**Comment by clalancette** [comment]
Closing in favor of https://github.com/kobuki-base/velocity_smoother/pull/1

**Pull request closed without merging**